src/shadow-embed.js

```diff
@@ -127,7 +127,8 @@ function createShadowRootPolyfill(hostElement) {
     // TODO(@dvoytenko) Consider to switch to a type union instead.
     /** @type {?}  */ (doc.createElement('i-amphtml-shadow-root')));
   hostElement.appendChild(shadowRoot);
-  hostElement.shadowRoot = hostElement.__AMP_SHADOW_ROOT = shadowRoot;
+  hostElement.__AMP_SHADOW_ROOT = shadowRoot;
+  Object.defineProperty(hostElement, 'shadowRoot', {value: shadowRoot});
```

Comment:
Might be worth noting that `Element.shadowRoot` is read-only and its being set here *because* its a polyfill.